networkx/algorithms/tree/branchings.py

```diff
@@ -363,147 +363,159 @@ def desired_edge(v):
 
             return edge, weight
 
-        while True:
-            # (I1): Choose a node v in G^i not in D^i.
-            try:
-               v = next(nodes)
-            except StopIteration:
-                # If there are no more new nodes to consider, then we *should*
-                # meet the break condition (b) from the paper:
-                #   (b) every node of G^i is in D^i and E^i is a branching
-                # Construction guarantees that it's a branching.
-                assert( len(G) == len(B) )
-                if len(B):
-                    assert( is_branching(B) )
-
-                if self.store:
-                    self.graphs.append(G.copy())
-                    self.branchings.append(B.copy())
+        def getFinalGraphAndBranch(nodes):
+            D = set([])
+            while True:
+                # (I1): Choose a node v in G^i not in D^i.
+                try:
+                   v = next(nodes)
+                except StopIteration:
+                    # If there are no more new nodes to consider, then we *should*
+                    # meet the break condition (b) from the paper:
+                    #   (b) every node of G^i is in D^i and E^i is a branching
+                    # Construction guarantees that it's a branching.
+                    assert( len(G) == len(B) )
```

Comment:
Assert statements don't need and typically don't have parentheses.

Also, if these are really things that are going to be checked, then maybe there's a more meaningful exception that can be raised here?